### ISSUE TITLE
[inputs.procstat] 

### ISSUE THREAD

**Issue by tguenneguez** [influxdata/telegraf#16206]
### Relevant telegraf.conf

```toml
[[inputs.procstat]]
  ## PID file to monitor process
  pid_file = "/opt/servicenow/mid/agent/work/midprod01.pid"
```

### Logs from Telegraf

```text
2024-09-18T15:07:44Z E! [inputs.procstat] Error in plugin: failed to read pidfile "/opt/servicenow/mid/agent/work/midprod01.pid": open /opt/servicenow/mid/agent/work/midprod01.pid: no such file or directory
```

### System info

Telegraf 1.32.1, Redhat 8.4

### Docker

N/A

### Steps to reproduce

1. Add process by pid file collect
2. Remove pid file
3. See the log in Error level

### Expected behavior

This use case is not illogical.
It therefore seems strange that with each collection in the absence of the pidFile a log line is added all the more in severity "Error".
This has the consequence of very quickly filling the FS or making important logs of other plugins disappear by rotation.
If the severity is appropriate, why in this case not generate an error when no process matches a filtering criterion?

I think the severity should be "Info".

### Actual behavior

An Error is generate for some normal use case.

### Additional info

The modification is here : https://github.com/influxdata/telegraf/blob/master/plugins/inputs/procstat/native_finder.go#L41
`	pidString, err := os.ReadFile(path)
	if err != nil {
		return pids, fmt.Errorf("failed to read pidfile %q: %w", path, err)
	}`
It should be :
`	pidString, err := os.ReadFile(path)
	if err != nil {
		return pids, nil
	}`

**Comment by srebhan** [comment]
@tguenneguez hmm why do you specify the PID file if it doesn't exist? Or is the error that the PID file is temporarily unavailable?

**Comment by tguenneguez** [comment]
If you specify a PID file (sample /var/run/sshd.pid), you monitor the process that have the pid in the file.
But when you stop this process, the file is remove.

It's a normal cas. pids is empty.

Maybe the severity of the message needs to be revised to the point of warning ?

**Comment by srebhan** [comment]
I agree. We could even make this a debug and only output the message as a warning once?!?

**Comment by tguenneguez** [comment]
Sorry for long reply.
I don't know how to create a debug severity message, but if I did, I'd do a PR.
For me, if the file doesn't exist, it should return an empty Pids list (no warning).
It's up to the centralization tool to notify this anomaly.

**Comment by tguenneguez** [comment]
Hello,
> We could even make this a debug and only output the message as a warning once?!?
This implies that the file state must be stored, and the message generated based on that state.
Thanks
